pkg/kubelet/prober/worker.go

```diff
@@ -222,6 +222,20 @@ func (w *worker) doProbe() (keepGoing bool) {
 			w.pod.Spec.RestartPolicy != v1.RestartPolicyNever
 	}
 
+	// Graceful shutdown of the pod.
+	if w.pod.ObjectMeta.DeletionTimestamp != nil && (w.probeType == liveness || w.probeType == startup) {
+		klog.V(3).Infof("Pod deletion requested, setting %v probe result to success: %v - %v",
+			w.probeType.String(), format.Pod(w.pod), w.container.Name)
+		if w.probeType == startup {
+			klog.Warningf("Pod deletion requested before successful container startup: %v - %v",
```

Comment:
nit for readability:

```suggestion
			klog.Warningf("Pod deletion requested before container has fully started: %v - %v",
```